### DISCUSSION TITLE
Recommandation for Go generated code

### DISCUSSION THREAD

**Discussion by ido-namely** [renovatebot/renovate#19435] [Request Help]
Hello,
I've noticed that Renovate has issues in most of our Go repositories which contains generated code (for example protobuf files) that is not checked in to our source control (Github).
This is not surprising, since `go mod tidy` command fails when it cannot find expected source files according to the dependency tree.

I was wondering if there is any recommendation on what to do in this case?
 - Are we forced to check in the generated code?
 - Would it be partially helpful if the `go mod tidy` command will be modified to `go mod tidy -e` to at least continue when it encounters error?
 - At this point I'm even considering writing a github action to be triggered after each PR to generate the code, run `go mod tidy` and push the changes to the Renovate branch.

Thanks in advance

**Comment by rarkins** [comment]
Do you self-host or using the hosted Mend Renovate app?

**Reply by ido-namely** [reply]
We use the Mend renovate app

**Reply by rarkins** [reply]
In this case it's a little harder because we still have limitations about running "arbitrary" scripts. But yes an action which runs on Renovate PRs would probably solve it for you. Either:
- commit as Renovate git author, so it doesn't notice. but in this case renovate might force push over it at any time so you need to make sure your action runs every time that happens, or
- commit as some other author but add it to gitIgnoredAuthors in renovate config

**Reply by ido-namely** [reply]
Thanks @rarkins ,
Is there a way then to prevent the failing artifact step?
I thought I could disable the gomod postupdate step, but that didn't seem to work.

Thanks!

**Reply by rarkins** [reply]
Which exact step fails? Is it only once it gets to `go mod tidy`, or is it the earlier `go get`? I thought `go mod tidy` was off by default, except perhaps for major updates which likely "break" anyway

**Reply by ido-namely** [reply]
You are right, It is the `go get` - 
`Command failed: docker run --rm --name=renovate_sidecar --label=renovate_child -v "/mnt/renovate/gh/namely/slug":"/mnt/renovate/gh/namely/slug" -v "/tmp/renovate-cache":"/tmp/renovate-cache" -v "/tmp/containerbase":"/tmp/containerbase" -e GOPATH -e GOPROXY -e GOFLAGS -e CGO_ENABLED -e GIT_CONFIG_KEY_0 -e GIT_CONFIG_VALUE_0 -e GIT_CONFIG_KEY_1 -e GIT_CONFIG_VALUE_1 -e GIT_CONFIG_KEY_2 -e GIT_CONFIG_VALUE_2 -e GIT_CONFIG_COUNT -e BUILDPACK_CACHE_DIR -e CONTAINERBASE_CACHE_DIR -w "/mnt/renovate/gh/namely/slug" docker.io/renovate/sidecar bash -l -c "install-tool golang 1.19.4 && go get -d -t ./..."`

Is there anything to do about that?

**Reply by rarkins** [reply]
Unfortunately you probably need #8804

**Reply by ido-namely** [reply]
yeah, that makes sense. So I suppose in the meantime there isn't an option to disable the `go get` command, just to keep it from indicating there is something wrong with the PRs (assuming we're going to go with our own post update task to remedy)

**Reply by rarkins** [reply]
That's not a configurable option but it would probably be reasonably easy to add, if you want to create a feature request
